PowerEditor/src/WinControls/ProjectPanel/TreeView.h

```diff
@@ -88,6 +112,29 @@ class TreeView : public Window {
 	void toggleExpandCollapse(HTREEITEM hItem) const {
 		TreeView_Expand(_hSelf, hItem, TVE_TOGGLE);
 	};
+
+	void cancelEdit() const {
+		TreeView_EndEditLabelNow(_hSelf, TRUE);
+	};
+
+	bool isExpanded(HTREEITEM hItem) const {
+		TVITEM tvItem;
+		tvItem.mask = TVIF_PARAM | TVIF_STATE;
+		tvItem.hItem = hItem;
+		::SendMessage(_hSelf, TVM_GETITEM, 0, (LPARAM)&tvItem);
+
+		return (tvItem.state & TVIS_EXPANDED) != 0;

```

Comment:
You ***absolutely*** should check to see if `::SendMessage` returned `FALSE` before using `tvItem.state`.